### PR TITLE
bpo-39573: Clean up modules and headers to use Py_IS_TYPE

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue39573

### CODE REVIEW

File: Objects/typeobject.c [hunk 1]
```diff
@@ -302,7 +302,7 @@ type_mro_modified(PyTypeObject *type, PyObject *bases) {
        each subclass when their mro is recursively updated.
      */
     Py_ssize_t i, n;
-    int custom = (Py_TYPE(type) != &PyType_Type);
+    int custom = Py_IS_TYPE(type, &PyType_Type);
```
Comment (vstinner) [inline Objects/typeobject.c]: This one looks wrong.

**Review by vstinner — CHANGES_REQUESTED** [review]
Please don't use git rebase, just add a commit, to ease review ;-) This PR is long to review.

File: Objects/typeobject.c (hunk 1 above)
Comment (corona10) [reply Objects/typeobject.c]: Oh.. good catch!

Comment (vstinner) [reply Objects/typeobject.c]: I know that you made this mistake on purpose. Just to check that I really review your PRs! :-)

Comment (corona10) [reply Objects/typeobject.c]: lol :)

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**